code/game/gamemodes/wizard/magic_tarot.dm

```diff
@@ -0,0 +1,885 @@
+	if(!COOLDOWN_FINISHED(src, card_cooldown))
+		. += "<span class='hierophant'>You may draw another card again in [round(COOLDOWN_TIMELEFT(src, card_cooldown) / 10)] second\s.</span>"
+
+/obj/item/tarot_generator/AltShiftClick(mob/user)
+	for(var/obj/item/magic_tarot_card/MTC in our_card_list)
+		MTC.dust()
+	to_chat(user, "<span class='hierophant'>You dispell the cards [src] had created.</span>")
+
+// Booster packs filled with 3, 5, or 7 playing cards! Used by the wizard space ruin, or rarely in lavaland tendril chests.
+/obj/item/tarot_card_pack
+	name = "\improper Enchanted Arcana Pack"
+	desc = "A pack of 3 Enchanted tarot cards. Collect them all!"
+	icon = 'icons/obj/playing_cards.dmi'
+	icon_state = "pack"
+	///How many cards in a pack. 3 in base, 5 in jumbo, 7 in mega
+	var/cards = 3
+
+/obj/item/tarot_card_pack/attack_self(mob/user)
+	user.visible_message("<span class='notice'>[user] tears open [src].</span>", \
+						"<span class='hierophant'>You tear open [src]!</span>")
+	playsound(loc, 'sound/items/poster_ripped.ogg', 50, TRUE)
+	var/prints = 0
+	while(prints < cards)
+		new /obj/item/magic_tarot_card(get_turf(src))
+		prints++
+	qdel(src)
+
+/obj/item/tarot_card_pack/jumbo
+	name = "\improper Jumbo Arcana Pack"
+	desc = "A Jumbo card pack from your friend Jimbo!"
+	icon_state = "jumbopack"
+	cards = 5
+
+/obj/item/tarot_card_pack/mega
+	name = "\improper MEGA Arcana Pack"
+	desc = "Sadly, you won't find a Joker for an angel room, or a Soul card in here either."
+	icon_state = "megapack"
+	cards = 7
+
+// Blank tarot cards. Made by the cult, however also good for space ruins potentially, where one feels a card pack would be too much?
+/obj/item/blank_tarot_card
+	name = "blank tarot card"
+	desc = "A blank tarot card."
+	icon = 'icons/obj/playing_cards.dmi'
+	icon_state = "tarot_blank"
+	w_class = WEIGHT_CLASS_TINY
+	throw_speed = 3
+	throw_range = 10
+	throwforce = 0
+	force = 0
+	resistance_flags = FLAMMABLE
+	///If a person can choose what the card produces. No cost if they can choose.
+	var/let_people_choose = FALSE
+
+/obj/item/blank_tarot_card/examine(mob/user)
+	. = ..()
+	if(!let_people_choose)
+		. += "<span class='hierophant'>With a bit of Ink, a work of art could be created. Will you provide your Ink?</span>"
+	else
+		. += "<span class='hierophant'>We have the Ink... Could you provide your Vision instead?</span>"
+
+/obj/item/blank_tarot_card/attack_self(mob/user)
+	if(!ishuman(user))
+		return
+	if(!let_people_choose)
+		var/mob/living/carbon/human/H = user
+		if(H.dna && (NO_BLOOD in H.dna.species.species_traits))
+			to_chat(user, "<span class='cult'>No blood to provide?...</span><span class='hierophant'> Then no Ink for the art...</span>")
+			return
+		if(H.blood_volume <= 100) //Shouldn't happen, they should be dead, but failsafe. Not bleeding as then they could recover the blood with blood rites
+			return
+		H.blood_volume -= 100
+		H.drop_item()
```

Comment:
Please put this higher and check if they can even drop it (With something like antidrop)